assignment2/task1.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: UTF-8 -*-
+
+
+class AncestorsFinder(object):
+    def __init__(self, graph):
+        self.graph = graph
+        self.root = 0
+        self.ancestors = None
+
+    def find(self, value):
+        self.ancestors = []
+        self._dfs(self.root, -1, value)
+        return self.ancestors
+
+    def _dfs(self, ver, prev, value):
+        if self.graph[ver]['value'] == value:
+            return True
+        for to in self.graph[ver]['edges']:
+            if to == prev or not self._dfs(to, ver, value):
+                continue
+            self.ancestors.append(self.graph[ver]['value'])
+            return True
+        return False
```

Comment:
Time complexity: O(n + m), where n - number of vertexes, m - number of edges in a graph